src/sentry/api/endpoints/organization_region.py

```diff
@@ -0,0 +1,73 @@
+from typing import Any
+
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import Endpoint, control_silo_endpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.permissions import SentryPermission
+from sentry.models.organizationmapping import OrganizationMapping
+from sentry.models.organizationmembermapping import OrganizationMemberMapping
+from sentry.types.region import get_region_by_name
+
+
+class OrganizationRegionEndpointPermissions(SentryPermission):
+    # Although this permission set is a bit weird, we need to have
+    # project:read for integration auth tokens, org:ci for org auth tokens
+    # and org:read for user auth tokens.
+    scope_map = {"GET": ["project:read", "org:ci", "org:read"]}
+
+    def has_object_permission(self, request, view, org_mapping: OrganizationMapping):
+        if request.auth is None or request.auth.organization_id is None:
+            assert not request.user.is_anonymous, "Anonymous users should not be missing auth"
```

Comment:
This will result in a 500. We could use `capture_message` here instead and return `False` as this is an invalid client request that we shouldn't respond to with a 500.